DopplerCustomDomain/CustomDomainProvider/TraefikRuleTypeEnum.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DopplerCustomDomain.CustomDomainProvider
+{
+    public enum TraefikRuleTypeEnum
+    {
+        HTTP_HTTPS,
+        HTTP_HTTPS_WITHOUT_REDIRECT,
+        HTTP
```

Comment:
Maybe

```suggestion
    public enum TraefikRuleTypeEnum
    {
        HttpsOnly,
        HttpsAndHttp,
        HttpOnly
```

Another option could be using EnumFlags:

```suggestion
    [Flags]
    public enum TraefikRuleTypeEnum
    {
        ListenHttp = 1,
        ListenHttps = 2, 
        RedirectHttpToHttps = 4
        HttpsStandard = ListenHttp | ListenHttps | RedirectHttpToHttps,
        HttpsAndHttp = ListenHttp | ListenHttps,
        HttpOnly = ListenHttp
```

Reply:
As we talk, it's not neccessary use Flags in this case.